src/EditorFeatures/CSharpTest/Completion/CompletionProviders/InternalsVisibleToCompletionProviderTests.cs

```diff
@@ -0,0 +1,140 @@
+
+using System;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Test.Utilities;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders
+{
+    public class InternalsVisibleToCompletionProviderTests : AbstractCSharpCompletionProviderTests
+    {
+        private static readonly string s_keyPairFile = SigningTestHelpers.KeyPairFile;
+        private static readonly DesktopStrongNameProvider s_defaultProvider = new SigningTestHelpers.VirtualizedStrongNameProvider(ImmutableArray.Create<string>());
+
+        public InternalsVisibleToCompletionProviderTests(CSharpTestWorkspaceFixture workspaceFixture) : base(workspaceFixture)
+        {
+            var ws = workspaceFixture.GetWorkspace();
+            var solution = ws.CurrentSolution;
+            var projectInfo1 = ProjectInfo.Create(ProjectId.CreateNewId(), VersionStamp.Create(), "ClassLibrary1", "ClassLibrary1", LanguageNames.CSharp,
+                compilationOptions: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, cryptoKeyFile: s_keyPairFile, strongNameProvider: s_defaultProvider));
+            var projectInfo2 = ProjectInfo.Create(ProjectId.CreateNewId(), VersionStamp.Create(), "ClassLibrary2", "ClassLibrary2", LanguageNames.CSharp);
+            var projectInfo3 = ProjectInfo.Create(ProjectId.CreateNewId(), VersionStamp.Create(), "ClassLibrary3", "ClassLibrary3", LanguageNames.CSharp,
+                compilationOptions: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, strongNameProvider: s_defaultProvider));
+            projectInfo3 = projectInfo3.WithMetadataReferences(new MetadataReference[] { MscorlibRef });
+            projectInfo3 = projectInfo3.WithDocuments(new DocumentInfo[] {
+                DocumentInfo.Create(DocumentId.CreateNewId(projectInfo3.Id), "AssemblyInfo.cs",
+                loader: TextLoader.From(TextAndVersion.Create(SourceText.From(
+                @"[assembly: System.Reflection.AssemblyKeyFile(""" + s_keyPairFile.Replace(@"\",@"\\") + @""")]"), VersionStamp.Default)))
+            });
+            var projectInfo4 = ProjectInfo.Create(ProjectId.CreateNewId(), VersionStamp.Create(), "ClassLibrary4", "ClassLibrary4", LanguageNames.CSharp,
+                compilationOptions: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, cryptoKeyFile: s_keyPairFile, strongNameProvider: s_defaultProvider, delaySign: true));
+            solution = solution.AddProject(projectInfo1).AddProject(projectInfo2).AddProject(projectInfo3).AddProject(projectInfo4);
+            ws.ChangeSolution(solution);
```

Comment:
so much easier to do if you just use CreateTestStateFromWorkspace and pass along a snippet of xml representing the workspace.

In general, we do not like manually creating workspaces in this manner.